File: src/2965. Find Missing and Repeated Values/soln.py
class Solution:
    def findMissingAndRepeatedValues(self, grid: List[List[int]]) -> List[int]:
        count = {}
        N = len(grid)
        for i in range(N):
            for j in range(N):
                if grid[i][j] not in count:
                    count[grid[i][j]] =0
                count[grid[i][j]] += 1
        
        double, missing = 0,0
        
        for num in range(1,N*N+1):
            if num not in count:
                missing = num
            elif count[num]==2:
                double =num
        return [double, missing]

        

File: src/2965. Find Missing and Repeated Values/soln.java
class Solution {
    public int[] findMissingAndRepeatedValues(int[][] grid) {

        int n = grid.length;

        int[] count = new int[n*n+1];

        for(int i=0;i<n;i++){
            for(int j=0;j<n;j++){
                count[grid[i][j]]++;
            }
        }

        int doubleNum = 0;
        int missingNum = 0;

        for(int num = 1; num<= n*n; num++){
            if(count[num] == 0){
                missingNum = num;
            }
            else if(count[num] == 2){
                doubleNum = num;
            }
        }
        
        return new int[]{doubleNum, missingNum};
    }
}
